indico/web/templates/forms/datetime_widget.html

```diff
@@ -21,9 +27,38 @@
 
             var dateField = $('#{{ field.id }}-date');
             var timeField = $('#{{ field.id }}-time');
+            var resetFieldAction = $('#{{ field.id }}-reset');
             var initialDate;
             var initialTime;
 
+            function clearDate() {
+                dateField.datepicker('setDate', null);
+                dateField.trigger('change');
+            }
+
+            function initTime() {
+                if (timeField.val() === '') {
+                    timeField.val({{ field.default_time | format_time('code', timezone=field.timezone) | tojson }});
+                }
+            }
+
+            function updateDateRequired(e) {
+                dateField.prop('required', e.target.value !== '');
+            }
+
+            function updateTimeRequired(e) {
+                timeField.prop('required', e.target.value !== '');
+            }
+
+            function updateResetFieldAction() {
+                var isVisible = dateField.val() !== '' || timeField.val() !== ''
```

Comment:
I don't mind changing it. I went for this because `isVisible` would hold the value and not a boolean.